lib/facter/resolvers/aix/disks.rb

```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+
+module Facter
+  module Resolvers
+    module Aix
+      class Disks < BaseResolver
+        @semaphore = Mutex.new
+        @fact_list ||= {}
+        class << self
+          private
+
+          MEGABYTES_EXPONENT = 1024**2
+          GIGABYTES_EXPONENT = 1024**3
+
+          def post_resolve(fact_name)
+            @fact_list.fetch(fact_name) { execute_lspv(fact_name) }
+          end
+
+          def execute_lspv(fact_name)
+            @fact_list[:disks] = {}
+
+            result = Facter::Core::Execution.execute('lspv', logger: log)
+
+            return if result.empty?
+
+            result.each_line do |line|
+              disk_name = line.split(' ')[0].strip
+              info = find_size(disk_name)
+              @fact_list[:disks][disk_name] = info if info
+            end
+
+            @fact_list[fact_name]
+          end
+
+          def find_size(name)
+            stdout = Facter::Core::Execution.execute("lspv #{name}", logger: log)
+
+            return if stdout.empty?
+
+            info_hash = extract_info(stdout)
+
+            return unless info_hash['PVSTATE']
+
+            size_bytes = compute_size(info_hash)
+
+            {
+              size_bytes: size_bytes,
+              size: Facter::FactsUtils::UnitConverter.bytes_to_human_readable(size_bytes)
+            }
+          end
+
+          def extract_info(lsl_content)
+            lsl_content = lsl_content.strip.split("\n").map do |line|
+              next unless /PP SIZE:|TOTAL PPs:|FREE PPs:|PV STATE:/ =~ line
+
+              line.split(/:|\s\s/)
```

Comment:
We could improve the regex so it does not generate so many empty value 
```
line.split(/:\s*|\s{2,}/)
```

Reply:
done